### ISSUE TITLE
Export waypoint with empty coords from PCN

### ISSUE THREAD

**Issue by murggel** [cgeo/cgeo#9338]
For a custom waypoint that has been configured to solve multicaches with the **waypoint calculator** using the **Plain** option, I can use c:geo to successfully write to the Personal Notes and upload them to GC.com, for example:

```
{c:geo-start}
@Formula (F) (NO-COORD)
"N50 57.(A+E)(B+B)(E) W001 21.(D)(B+D)(A+B+B)
telephone number 
023806ABCDE
short walk
Checksum 30"
{c:geo-end}
```

Which is fantastic but when using c:geo on another device and downloading this cache, my expectation was that the same custom waypoint would be recreated ready for action in the field.

Unfortunately, the read process fails to populate this waypoint. I am guessing the reason is because the coordinates are not valid, that is, they are in the format of a valid formula format for the waypoint latitude and longitude but not one of the excepted valid coordinate formats:

```
Plain
N50 57.(A+E)(B+B)(E)
W001 21.(D)(B+D)(A+B+B)
A=|a+b| D=|a+b|
B=|a+b| E=|a+b|
Note ...
```

I see this style of waypoint calculations to be the most common usage, so it will be a shame if this cannot be automated to create this waypoint on load rather than going through copy/paste manual configurations, which is doable ... but a pain ...

If this was fixed, we could also do our planning in the cache page, updating the Personal Notes in the format template:

```
{c:geo-start}
@Title (F) (NO-COORD)
"Valid Lat/Long Formula for calculator
additional notes"
{c:geo-end}
```

Or in whatever format that works best for the WPC and WP authors.

Then, we load our GPX into c:geo and are good to go.

_Originally posted by @Billy-Wallace in https://github.com/cgeo/cgeo/issues/9234#issuecomment-713897472_

-> see #9211 for parsing formula

this issue is for extract user waypoint with empty coordinates from PCN (it will be already written to PCN, but no waypoint is created).
Advantage: no dummy-coordinates have to be added to the waypoint.

**Comment by murggel** [comment]
I can't add the "Feature Request"-label. 
Can anyone please assign it?

**Comment by Billy-Wallace** [comment]
After doing some field testing and better understanding the process myself, I would like to show a couple of examples that will hopefully help the developers better understand what I believe would be a great additional feature for c:geo.

### Current Personal Notes to Calculated Waypoint Process

Currently, the best a user can do in preparing c:geo to assist with solving multicaches is to update the Personal Notes on the cache page in the format required by c:geo to add a waypoint. A valid set of coordinates are required, so the published cache coordinates are probable the best option:

```
{c:geo-start}
@Calculate Finals (F) N50 27.555 W001 39.000
"N50 27.ABC W001 39.(A-B)D(E+B)
Number on Telegraph Pole = ABCDE, plus any other useful notes"
{c:geo-end}
```

This succesfully creates the waypoint on import into c:geo. The user is then left with the task of changing this waypoint to a 'Plain' waypoint and then copy/paste the latitude and longitude parts from the formula into the respective fields (_Very fiddley out there in the sticks_).

If there is more than one member in the team, each user has to go through this same process.

Each time the cache is refreshed or re-imported into c:geo, the process has to be repeated all over again.

### Feature Enhancement for PN to CWP Process

To resolve these issues and make this very useful for all c:geo users, would be to replace the valid coordinates with a valid formula as follows:

```
{c:geo-start}
@Calculate Finals (F) N50 27.ABC W001 39.(A-B)D(E+B)
"Number on box = ABCDE, plus any other useful notes"
{c:geo-end}
```

The process would need to be updated to check the formula is valid and then create a 'Plain' waypoint with the latitude and longitude copied into the lat/long fields.

This will mean no copy/paste preparation is required and the user can directly enter values into the variables out in the field.

Of course, scraping the cache page for the formula to automate creation of calculated waypoints is probably asking too much. 

But I do like the ability of creating a note for the waypoint that is minimal and saves having to refer back to the cache description. Which as we all know, can sometimes be difficult to read in long, content rich cache descriptions. Reading the questions and entering the values is all focused in one place. _The ultimate solution_.

**Comment by eddiemuc** [comment]
Hi all,
while I don't know calculated Waypoints very much I do know the parsing process of Personal Notes into waypoints and vice versa.

I believe that a format like the following could be included quite well into the current parsing process (@Billy-Wallace I used your proposal as a starting point but modified it a bit to fit better to the current parsing process):

```
{c:geo-start}
@Calculate Finals (F) (CC:N50 27.ABC W001 39.(A-B)D(E+B))
"Number on box = ABCDE, plus any other useful notes"
{c:geo-end}
```

Note that I surrounded the actual coordinate formula with `(CC:...)`, which could be the parsing ancor for c:geo to know that this shall be a waypoint with calculated coordinates. (CC stands for Calculated Coordinate, of course a different token can be used here)

Now parsing calculated coordinates from the cache listing would be a whole different challenge... :-)

**Comment by murggel** [comment]
Hi all,
 I already looked into it, because I had nearly the same thoughts as @Billy-Wallace .
I could imagine something like (see #9211)
```
{c:geo-start}
@Calculate Finals (F) (FORMULA)
N50 27.ABC W001 39.(A-B)D(E+B))
A=|a+b| D=|a+b|
B=|a+b| E=|a+b|
"Number on box = ABCDE, plus any other useful notes"
{c:geo-end}
```
So the whole formula can be added to the note. Here I don't know at the moment how to separate the expressions. 
With some help I would like to give this a try.

To avoid having valid coordinates for right now, I prepared the PR with the ability to extract the following, so the coordinates kept empty.
```
{c:geo-start}
@Calculate Finals (F) (NO-COORDS)
"N50 27.ABC W001 39.(A-B)D(E+B))
Number on box = ABCDE, plus any other useful notes"
{c:geo-end}
```

Sorry, my english is not so well, and I am not so good in writing long posts :-)
